subsys/bluetooth/mesh/beacon.c

```diff
@@ -96,12 +96,15 @@ static bool secure_beacon_send(struct bt_mesh_subnet *sub, void *cb_data)
 	uint32_t now = k_uptime_get_32();
 	struct net_buf *buf;
 	uint32_t time_diff;
+	uint32_t time_diff_observed;
 
 	BT_DBG("");
 
 	time_diff = now - sub->beacon_sent;
+	time_diff_observed = now - sub->beacon_recv;
 	if (time_diff < (600 * MSEC_PER_SEC) &&
-		time_diff < BEACON_THRESHOLD(sub)) {
+		(time_diff < BEACON_THRESHOLD(sub) ||
+		 time_diff_observed < (10 * MSEC_PER_SEC))) {
```

Comment:
This will cause us to wait up to 20 seconds before sending our beacon, if the timing aligns just right. Will this be reasonable in topologies like:

    A - us - B

where node A keeps sending a beacon every 10 seconds, and we end up backing down, causing B to never receive a beacon? Unless node A receives beacons from some other device we can't see, we'll virtually never send beacons in this case, since it will always be less than 10 seconds since we last saw A's beacon. Will this end up being unrecoverable for us? Or does this happen already with the existing beacon threshold approach?

@alxelax please weigh in when you have time :)

Reply:
This should not happen. 'time_diff' will keep growing in that case. Beacon will be forcefully sent at least every 600 seconds by the receiver which is backing off continuously (Line 105).

Reply:
Yep, @omkar3141 is right. If there is device A that spams with a beacon every 10 seconds then 'our' device will be able to send a beacon at least once in 600 seconds. 

The current implementation sends a beacon if `BEACON_THRESHOLD` expired on the next 10s interval. 

However:
```
3.9.3.1 Secure Network beacon behavior 
The expected behavior is that each node receives one beacon for a given subnet approximately every 10 seconds.
```
but in this topology B will receive a beacon about every 600 seconds. Hm

Reply:
Since radio transmission is omnidirectional in nature, this recommendation for each and every node will be impossible to follow.